File: FileLibrary/src/main/java/com/xx/Filelibrary/OfficeFileAdapter.java
package com.xx.Filelibrary;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;


import java.io.File;
import java.util.List;

public  class OfficeFileAdapter extends RecyclerView.Adapter<OfficeViewHolder> {
    private  Context context;
    private  List<File> list;

    public OfficeFileAdapter(Context c, List<File> l) {
        this.context = c;
        this.list = l;
    }


    @NonNull
    @Override
    public OfficeViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_txt, parent, false);

        return new OfficeViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull OfficeViewHolder holder, int position) {
        holder.tv.setText(list.get(position).getName());
    }

    @Override
    public int getItemCount() {
        return list.size();
    }




}

File: FileLibrary/src/main/java/com/xx/Filelibrary/OfficeFileAdapter.kt
package com.xx.Filelibrary

import android.content.Context
import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import com.bumptech.glide.request.RequestOptions
import java.io.File

class OfficeFileAdapter(c: Context, l: List<File>):RecyclerView.Adapter<OfficeViewHolder>() {
    val context:Context=c;
    val list:List<File> = l

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OfficeViewHolder {

        val view: View =LayoutInflater.from(context).inflate(R.layout.item_txt, parent, false);

        return OfficeViewHolder(view);
    }

    override fun onBindViewHolder(holder: OfficeViewHolder, position: Int) {

        holder.tv.setText(list.get(position).name)
    }

    override fun getItemCount(): Int {
        return list.size
    }
}